src/main/java/com/wanderersoftherift/wotr/effects/CriticalEffect.java

```diff
@@ -0,0 +1,21 @@
+package com.wanderersoftherift.wotr.effects;
+
+import com.wanderersoftherift.wotr.init.WotrAttributes;
+import net.minecraft.util.RandomSource;
+import net.minecraft.world.entity.LivingEntity;
+
+public class CriticalEffect {
+    public static float calcFinalDamage(
+            float initialDamage,
+            LivingEntity attacker,
+            LivingEntity foe,
+            RandomSource random) {
+
+        int critChance = (int) attacker.getAttributeValue(WotrAttributes.CRITICAL_CHANCE)
+                - (int) foe.getAttributeValue(WotrAttributes.CRITICAL_AVOIDANCE);
+        float critMult = (float) attacker.getAttributeValue(WotrAttributes.CRITICAL_MULTIPLIER);
+        int critApplications = (critChance + random.nextInt(100)) / 100;
+
+        return initialDamage * (critMult * critApplications);
```

Comment:
Why did you undo the change that I made?
```suggestion
        return initialDamage * ((critMult - 1) * critApplications + 1);
```
Final damage for damage multiplier 1.5:
| applications | current version | my version|
| -- | -- | -- |
| 0 | 0× | 1× |
| 1 | 1.5× | 1.5× |
| 2 | 3× | 2× |

We don't want 0 damage with 0 applications.

Maybe multiplier should be replaced with "bonus" that would start at 50%,. That would be effectively same as my version.
```suggestion
        return initialDamage + initialDamage * critBonus * critApplications;
```

Reply:
In the second variation, the `critBonus` is *extra* damage on top of the base damage. For to get `critMiltiplier=1.5`, the bonus would be `0.5`. In general `critBonus = critMultiplier - 1`. And the first variation is the same, just written differently.